Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -884,6 +884,19 @@ void TextEditor::keydown_event(KeyEvent& event)
     event.ignore();
 }
 
+void TextEditor::delete_previous_word()
+{
+    TextPosition word_break = document().first_word_break_before(m_cursor, true);
+    TextRange to_erase(word_break, m_cursor);
+
+    if (document().text_in_range(to_erase).trim_whitespace().length() == 0) {
+        word_break = document().first_word_break_before(word_break, true);
+        to_erase.set_start(word_break);
+    }
```

Comment:
Shouldn't `first_word_break_before` skip the whitespace correctly the first time?

Reply:
As in, I should move this logic into `first_word_break_before` itself?

Reply:
Perhaps a `first_word_before` for `TextDocument` could be nicer. I conflated `word_break` with `word`, my bad.